src/payment/PaymentPage.jsx

```diff
@@ -169,8 +171,9 @@ class PaymentPage extends React.Component {
             ),
           }}
         />
-        {loaded && isEmpty ? this.renderEmptyMessage() : null}
-        {loading || !isEmpty ? this.renderBasket() : null}
+        {(!loaded || loading || isValidBasket) ? this.renderBasket() : null}
```

Comment:
What do you think of making a descriptively-named variable for the conditional part of this?  On the face of it, its not clear reading this why if it's loading we render the basket, but if it's a redirect we render loading.

Reply:
Admittedly the names are getting a bit fuzzy.  I agree.  🤔

Reply:
I tried to clean it up!